### DISCUSSION TITLE
Add standalone binary to releases

### DISCUSSION THREAD

**Discussion by thiagomajesk** [dcastil/tailwind-merge#200] [Ideas]
Hi @dcastil! Thanks for this fantastic project.

I was wondering if you would be open to creating a standalone cli to tailwind-merge like the one we have for Tailwind: https://github.com/tailwindlabs/tailwindcss/tree/master/standalone-cli. 

I think there are some use cases outside the js ecosystem where having a tool like this embedded would be very useful. For instance, I'm trying to create a design system using Tailwind for another language and I currently don't have good alternatives to tailwind-merge that I can use to deal with merge conflicts.

It seems that it would be very straightforward to add this to the project. What do you say?

**Comment by dcastil** [comment]
Hey @thiagomajesk! 👋

That's a very interesting idea. Currently I don't have the capacity to build this, but maybe I can do this in the future. Or would you like to submit a PR?

I didn't expect that tailwind-merge had uses beyond the browser. Can you tell me more about your use case? Would you like to use the binary during the build process of the design system or during the runtime of a server? Which language are you using for the design system?

**Reply by thiagomajesk** [reply]
Hi @dcastil, I'm currently building a component library for Elixir/Phoenix so it would be cool to have a way to reduce conflicting styles. 

I thought about submitting a PR but I wanted to check with you first. I'm currently doing some exploration with the public API for this library and deciding what is going to be the best approach. As soon as I'm done with I can test some solutions for this.

BTW, our ecosystem already has an embedded Tailwind package that uses the standalone CLI. If you want to take a look, this is the repo: https://github.com/phoenixframework/tailwind.

> I didn't expect that tailwind-merge had uses beyond the browser

From my experience so far, merging classes in a non-conflicting way is such a common use case that I was amazed that this isn't part of the standard tailwind cli yet. BTW, the TW vs code extension does some linting for conflicting classes, have you thought about collaborating with them on this front?

**Reply by dcastil** [reply]
> I'm currently building a component library for Elixir/Phoenix so it would be cool to have a way to reduce conflicting styles.

> BTW, our ecosystem already has an embedded Tailwind package that uses the standalone CLI.

Ah yeah, that makes sense! Someone actually ported tailwind-merge to Elixir here https://github.com/bratsche/twix. But I see that you starred it already, so you probably already know about it. 😁

> I thought about submitting a PR but I wanted to check with you first. I'm currently doing some exploration with the public API for this library and deciding what is going to be the best approach. As soon as I'm done with I can test some solutions for this.

Yeah if you have a rough idea how it could look like, you can create a draft PR with a prototype to start a discussion. If the setup is straight forward and the maintenance surface area isn't too high, I'm happy to add this to the package!

> BTW, the TW vs code extension does some linting for conflicting classes, have you thought about collaborating with them on this front?

So far not much. I have an open discussion in https://github.com/tailwindlabs/tailwindcss/discussions/10348 for better API interop and there are people [here and there](https://github.com/tailwindlabs/tailwindcss/discussions/1446#discussioncomment-4459791) requesting more integration between the two. Let's see where this will lead to. 😊
